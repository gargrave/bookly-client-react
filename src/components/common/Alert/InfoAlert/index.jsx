import React from 'react'
import { string } from 'prop-types'

import { prependClass } from '@/utils/cssHelpers'

const classes = () => {
  return `${prependClass('alert')} alert alert-info`
}

const InfoAlert = ({
  message,
}) => (
  <div className={classes()}>
    {message}
  </div>
)

InfoAlert.propTypes = {
  message: string.isRequired,
}

export default InfoAlert
