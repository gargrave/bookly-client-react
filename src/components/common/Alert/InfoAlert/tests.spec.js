import React from 'react'
import { shallow } from 'enzyme'

import InfoAlert from './'

describe('InfoAlert', () => {
  it('renders correctly with an error message', () => {
    const message = 'OMG THIS IS A ERROR'
    const component = shallow(<InfoAlert message={message} />)
    const element = component.find('.bookly-alert.alert.alert-info')
    expect(component).toMatchSnapshot()
    expect(element.length).toEqual(1)
    expect(element.text()).toEqual(message)
  })
})
