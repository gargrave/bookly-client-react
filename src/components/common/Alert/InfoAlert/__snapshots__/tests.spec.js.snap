// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoAlert renders correctly with an error message 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="bookly-alert alert alert-info"
>
    OMG THIS IS A ERROR
</div>,
  "nodes": Array [
    <div
      className="bookly-alert alert alert-info"
>
      OMG THIS IS A ERROR
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InfoAlert
        message="OMG THIS IS A ERROR"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "message": "OMG THIS IS A ERROR",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="bookly-alert alert alert-info"
>
          OMG THIS IS A ERROR
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="bookly-alert alert alert-info"
>
          OMG THIS IS A ERROR
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InfoAlert
    message="OMG THIS IS A ERROR"
/>,
}
`;
